iroh-willow/src/engine/actor.rs

```diff
@@ -201,7 +202,21 @@ impl Drop for ActorHandle {
         // this means we're dropping the last reference
         if let Some(handle) = Arc::get_mut(&mut self.join_handle) {
             let handle = handle.take().expect("can only drop once");
-            self.inbox_tx.send(Input::Shutdown { reply: None }).ok();
+
+            match tokio::runtime::Handle::try_current() {
+                Ok(runtime) => {
+                    let (dumb, _) = tokio::sync::mpsc::channel(1);
+                    let inbox_tx = std::mem::replace(&mut self.inbox_tx, dumb);
+                    runtime
+                        .spawn(async move { inbox_tx.send(Input::Shutdown { reply: None }).await });
+                }
+                Err(_) => {
+                    self.inbox_tx
+                        .blocking_send(Input::Shutdown { reply: None })
+                        .ok();
+                }
+            }
```

Comment:
@Frando obviously this is quite a workaround for the fact that `blocking_send` panics if there's a runtime.

We should either move this into some central utility, or just switch to `async_channel` instead (like most other places, tbh).
But kind of good to know that an MPSC is technically good enough for us.

Reply:
Yeah. I don't have a better solution OTOH. @rklaehn fought this hard as well.